src/library_nodefs.js

```diff
@@ -127,6 +127,11 @@ addToLibrary({
         if (NODEFS.isWindows && !stat.blocks) {
           stat.blocks = (stat.size+stat.blksize-1)/stat.blksize|0;
         }
+        if (NODEFS.isWindows) {
+          // Node.js on Windows never represents permission bit 'x', so
+          // propagate read bits to execute bits.
+          stat.mode = stat.mode | (stat.mode & 292) >> 2;
```

Comment:
Can you put brackets around the `(stat.mode & 292) >> 2` so its clear that the shift only applies the part of this expression?

Reply:
I noticed you updated the code where this is based in https://github.com/emscripten-core/emscripten/pull/21904. I'll apply the same here. Thanks 🙇 !

Reply:
Done in https://github.com/emscripten-core/emscripten/pull/21902/commits/75abe47bf4c8a12f0fb3d8ed7d1bf70518bb9b21.